tests/test_filtering.py

```diff
@@ -0,0 +1,87 @@
+from kbmod import *
+import numpy as np
+import tempfile
+import unittest
+
+class test_kernels_wrappers(unittest.TestCase):        
+    def test_sigmag_filtered_indices_same(self):
+        # With everything the same, nothing should be filtered.
+        values = [1.0 for _ in range(20)]
+        inds = sigmag_filtered_indices(values, 0.25, 0.75, 0.7413);
+        self.assertEqual(len(inds), 20)
+        for idx in inds:
+            self.assertGreaterEqual(values[idx], -1.0)
+            self.assertLessEqual(values[idx], 1.0)
+
+    def test_sigmag_filtered_indices_no_outliers(self):
+        # Try with a median of 1.0 and a percentile range of 3.0 (2.0 - -1.0).
+        # It should filter any values outside [-3.45, 5.45]
+        values = [-1.0, -1.0, -1.0, 0.0, 1.0, 2.0, 2.0, 2.0, 3.1];
+        inds = sigmag_filtered_indices(values, 0.25, 0.75, 0.7413)
+        self.assertEqual(len(inds), len(values))
+
+    def test_sigmag_filtered_indices_one_outlier(self):
+        # Try with a median of 1.0 and a percentile range of 3.0 (2.0 - -1.0).
+        # It should filter any values outside [-3.45, 5.45]
+        values = [-1.0, -1.0, -1.0, 0.0, 1.0, 2.0, 2.0, 2.0, 5.46];
+        inds = sigmag_filtered_indices(values, 0.25, 0.75, 0.7413)
+        self.assertEqual(len(inds), len(values) - 1)
+
+        # The only missing value should be index=8.
+        for i in range(8):
+            self.assertTrue(i in inds)
+        self.assertFalse(8 in inds)
+
+    def test_sigmag_filtered_indices_other_bounds(self):
+        # Do the filtering of test_sigmag_filtered_indices_one_outlier
+        # with wider bounds [-5.232608, 7.232608].
```

Comment:
This seems a little off to me. I would think the 85th percentile value should be 2 not 5.46 so that the bounds should be [-1.89, 3.89]. Is this a bug in the percentile value selection? Or is this a result of the percentile calculation working the way it should but in a different way than I expect?

Reply:
This is a really good catch. I've been using the percentile computation from the original code:
`int((num_values + 1) * sGL1 + 0.5) - 1`
but I think a better calculation would be to use:
`int(num_values * sGL1)`
Does that sound right to you?

Reply:
Actually, after looking it up. It sounds like a common approximation for the index is:
`ceil(num_values * sGL1)`
To avoid problems with floating point precision and account for being zero indexed, we can use:
`int(ceil(num_values * 0.5) + 0.001) - 1;`

Reply:
Good call. I added checks to the C++ function and then additional clipping in the on GPU function.